newIDE/app/src/EventsSheet/ParameterFields/LeaderboardIdField.js

```diff
@@ -0,0 +1,147 @@
+// @flow
+import { Trans } from '@lingui/macro';
+import { type ParameterInlineRendererProps } from './ParameterInlineRenderer.flow';
+import React from 'react';
+import { type ParameterFieldProps } from './ParameterFieldCommons';
+import SelectField from '../../UI/SelectField';
+import SelectOption from '../../UI/SelectOption';
+import { TextFieldWithButtonLayout } from '../../UI/Layout';
+import RaisedButton from '../../UI/RaisedButton';
+import { type Leaderboard } from '../../Utils/GDevelopServices/Play';
+import LeaderboardContext from '../../Leaderboard/LeaderboardContext';
+import OpenInNew from '@material-ui/icons/OpenInNew';
+import { t } from '@lingui/macro';
+import Toggle from '../../UI/Toggle';
+import { LargeSpacer } from '../../UI/Grid';
+import LeaderboardDialog from '../../SceneEditor/LeaderboardDialog';
+import GenericExpressionField from './GenericExpressionField';
+
+const getLeaderboardNameByValue = (
+  leaderboards: ?Array<Leaderboard>,
+  value: string
+): string => {
+  if (!leaderboards) return value;
+  const quotedIdLeaderboards = leaderboards.map(leaderboard => ({
+    id: `"${leaderboard.id}"`,
+    name: leaderboard.name,
+  }));
+  const leaderboard = quotedIdLeaderboards.find(
+    leaderboard => leaderboard.id === value
```

Comment:
To avoid having to define a new variable `quotedIdLeaderboards`, your test here could be:

```js
leaderboards.find( leaderboard => `"${leaderboard.id}"` === value )
`
``

Reply:
I see you did that just below, maybe there's something I'm missing

Reply:
No you are right, since we return only the name when found, we do not need to create quotedIdLeaderboards